docs/dev/framework/security.md

```diff
@@ -163,6 +171,61 @@ $security->isGranted(ContaoNewsPermissions::USER_CAN_CREATE_ARCHIVES);
 {{% /notice %}}
 
 
+### Example
+
+By default admins can access everything and you can restrict access to back end sections only for non-admins via back end user groups. The 
+following example implements a custom voter for your application which grants access to the "Maintenance" back end section only for the 
+admin with ID "1".
+
+```php
+// src/Security/Voter/AdminMaintenanceAccessVoter.php
+namespace App\Security\Voter;
+
+use Contao\BackendUser;
+use Contao\CoreBundle\Security\ContaoCorePermissions;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+use Symfony\Component\Security\Core\Security;
+
+class AdminMaintenanceAccessVoter extends Voter
+{
+    private $security;
+
+    public function __construct(Security $security)
+    {
+        $this->security = $security;
+    }
+
+    protected function supports(string $attribute, $subject)
+    {
+        // Abstain, if we are not voting for maintenance back end module access
+        if ('maintenance' !== $subject || $attribute !== ContaoCorePermissions::USER_CAN_ACCESS_MODULE) {
+            return false;
+        }
+
+        // Get the currently logged in user
+        $user = $this->security->getUser();
+
+        // Abstain, if not back end admin
+        if (!$user instanceof BackendUser || !$user->isAdmin) {
```

Comment:
Imho that's wrong. I thought you should be checking roles? /cc @aschempp

Reply:
If I need to access the user object anyway, then it should not matter, should it?

Reply:
Besides, Contao's `BackendAccessVoter` does the same:

https://github.com/contao/contao/blob/3ce5a595d30a4cb26361f03f368f9bf38faa560e/core-bundle/src/Security/Voter/BackendAccessVoter.php#L65-L67